analysis_driver/dataset.py

```diff
@@ -403,6 +433,31 @@ def number_of_samples(self):
                 raise AnalysisDriverError('Could not find number of quoted samples in LIMS for ' + self.name)
         return self._number_of_samples
 
+    @property
```

Comment:
Yes, but arguably better than doing a `pop` on a set of len > 1 and not knowing which species you'll get. This has always been the case anyway - the logic has just been moved to ProjectDataset, not actually changed. We can fix that later in a different branch if we need to support multi-species projects.